src/message_dao_test.py

```diff
@@ -0,0 +1,79 @@
+import unittest
+import subprocess
+import zlib
+
+from unittest import mock
+from google.cloud.sql.connector import Connector
+
+import message
+import message_dao
+import archive_converter
+from test_helpers import test_data_path
+
+class MessageDaoTest(unittest.TestCase):
+
+    def setUp(self):
+        self.addCleanup(mock.patch.stopall)
+        self.mock_connect = mock.patch.object(Connector, 'connect').start()
+        mock_connection = self.mock_connect.return_value
+        self.mock_commit = mock_connection.commit
+        self.mock_cursor = mock_connection.cursor.return_value.__enter__.return_value
+        self.mock_execute = self.mock_cursor.execute
+        # Check initialization of DAO
+        self.dao = message_dao.MessageDao('FAKE_GIT_PATH')
+        self.mock_connect.assert_called_once()
+        self.mock_commit.assert_called_once()
+        self.assertEqual(3, self.mock_execute.call_count)
+        self.mock_connect.reset_mock()
+        self.mock_execute.reset_mock()
+        self.mock_commit.reset_mock()
+        self.mock_connect.side_effect = RuntimeError("Shouldn't be called after init")
+
+    def test_store(self):
+        email = archive_converter.generate_email_from_file(test_data_path('patch6.txt'))
+        sql_text = "REPLACE INTO Messages VALUES (%s, %s, %s, %s, %s, %s, %s)"
+        self.dao.store(email)
+        self.mock_execute.assert_called_once_with(sql_text, mock.ANY)
+        self.mock_commit.assert_called_once()
+
+    @mock.patch.object(subprocess, 'check_output')
+    @mock.patch.object(message_dao, 'parse_message_from_str')
+    def test_get(self, mock_parse_msg, mock_check_output):
+        self.mock_commit.side_effect = RuntimeError("Shouldn't be called during get")
+        email = archive_converter.generate_email_from_file(test_data_path('patch6.txt'))
+        mock_parse_msg.return_value = email
+        self.mock_cursor.fetchone.return_value = (email.archive_hash, email.change_id)
+        self.assertEqual(email, self.dao.get('fake_message_id'))
+        self.assertEqual(2, self.mock_execute.call_count) 
```

Comment:
I already forgot and got confused why this would be 2.
We could add a comment explaining _get_children() is called to.

Or if we want, we could also do

```python
class StrContains(string):
   def __eq__(self, other):
      return self in other
...

self.assertEqual(2, self.mock_execute.call_count)
self.mock_execute.assert_has_calls([
  mock.call(StrContains("where message_id=%s"), mock.ANY),  # get this msg
  mock.call(StrContains("where in_reply_to=%s"), mock.ANY),  # get children
 ])
 ```

Reply:
Clever solution! Modified to utilize StrContains in addition to asserting call counts: that way it's explicit about where the execute methods calls are coming from.